src/schema/schema.rs

```diff
@@ -169,12 +173,30 @@ pub struct Schema<Root: SchemaRoot> {
     pub transitions: TinyOrdMap<TransitionType, TransitionSchema>,
 
     /// Type system
-    // TODO: Commit to type system id, not the type system itself
     pub type_system: TypeSystem,
     /// Validation code.
     pub script: Script,
 }
 
+impl<Root: SchemaRoot> CommitEncode for Schema<Root> {
+    type CommitmentId = SchemaId;
+
+    fn commit_encode(&self, e: &mut CommitEngine) {
+        e.commit_to(&self.ffv);
+        e.commit_to(&self.subset_of.as_ref().map(Root::schema_id));
+
+        e.commit_to(&self.global_types);
+        e.commit_to(&self.owned_types);
+        e.commit_to(&self.valency_types);
+        e.commit_to(&self.genesis);
+        e.commit_to(&self.extensions);
+        e.commit_to(&self.transitions);
+
+        e.commit_to(&self.type_system.id());
+        e.commit_to(&self.script);
```

Comment:
I think all types need be named type, otherwise, the error will occur: 

`commit encoder can commit only to named types`